File: TwitterPage/Common/DataModels/Network/TweetModel.m
//
//  TweetModel.m
//  TwitterPage
//
//  Created by DMITRY SINYOV on 30.08.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

#import "TweetModel.h"
#import "ParseFieldChecker.h"
#import "CommonDateFormatter.h"

@implementation TweetModel

-(instancetype)initWithJSON:(NSDictionary*)JSON {
    self = [super initWithJSON:JSON];
    
    self.tweetID = [ParseFieldChecker checkStringOrDecimalNumberToString:JSON[@"id"]];
    self.text = [ParseFieldChecker fieldDataForJSONfield:JSON[@"text"]];
    
    NSString *stringDate = [ParseFieldChecker fieldDataForJSONfield:JSON[@"created_at"]];
    if (stringDate) self.date = [CommonDateFormatter dateFromServerString:stringDate];
    
    NSNumber *favorited = [ParseFieldChecker checkStringOrDecimalNumberToNumber:JSON[@"favorited"]];
    if (favorited) self.favorited = favorited.boolValue;
    
    NSDictionary *userDict = [ParseFieldChecker fieldDataForJSONfield:JSON[@"user"]];
    
    if (userDict) self.user = [[UserModel alloc] initWithJSON:userDict];
    
    if (!self.tweetID || !self.user || self.user.isIncorrect) self.isIncorrect = YES;
    
    return self;
}

@end

File: SWIFT/TwitterPageSWIFT/Common/DataModels/Network/TweetModel.swift
//
//  TweetModel.swift
//  TwitterPageSWIFT
//
//  Created by DMITRY SINYOV on 15.09.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

import UIKit

class TweetModel: CommonModel {
    
    var tweetID:String? = nil
    var text:String? = nil
    var date:Date? = nil
    var favorited = false
    var user:UserModel? = nil
    
    override init(json: [String: AnyObject]) {
        super.init(json: json)
        
        self.tweetID = ParseFieldChecker.checkStringOrDecimalNumberToString(checkedObject: json["id"])
        
        self.text = ParseFieldChecker.fieldDataForJSON(field: json["text"] as? String)
        
        let stringDate = ParseFieldChecker.fieldDataForJSON(field: json["created_at"] as? String)
        if (stringDate != nil) { self.date = CommonDateFormatter.dateFromServerString(serverString: stringDate!) }
        
        let favorited = ParseFieldChecker.checkStringOrDecimalNumberToNumber(checkedObject: json["favorited"])
        if (favorited != nil) { self.favorited = favorited!.boolValue }
        
        let userDict = ParseFieldChecker.fieldDataForJSON(field:json["user"])
        
        if (userDict != nil) { self.user = UserModel(json: userDict as! [String: AnyObject]) }
        
        if (self.tweetID == nil || self.user == nil || (self.user != nil && self.user!.isIncorrect)) { self.isIncorrect = true }
    }

}
